pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -583,22 +541,22 @@ func runPrivilegeChecks(
 }
 
 // runSinglePrivilegeCheck runs the provided check function for the privilege.
-// If the privilege has the grantOption flag set to true, it also runs the
+// If the privilege has the GrantOption flag set to true, it also runs the
 // provided function with the GRANT privilege and only returns True if both
-// calls returns True. See the comment on priv for justification.
+// calls returns True. See the comment on Priv for justification.
 func runSinglePrivilegeCheck(
-	priv priv, check func(privilege.Kind) (tree.Datum, error),
+	priv privilege.Priv, check func(privilege.Priv) (tree.Datum, error),
 ) (tree.Datum, error) {
-	d, err := check(priv.kind)
+	d, err := check(priv)
 	if err != nil {
 		return nil, err
 	}
 	switch d {
 	case tree.DBoolFalse, tree.DNull:
 	case tree.DBoolTrue:
-		if priv.grantOption {
-			// grantOption is set, so AND the result with check(GRANT).
-			d, err = check(privilege.GRANT)
+		if oldVersion { //todo
```

Comment:
What is the best way to get this context from here?
`p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.DefaultPrivileges)`

Reply:
hm yeah, we don't have a `planner` here. let me think about it. perhaps the version gate needs to be done deeper in the stack?